2/contrib/s2i/run

```diff
@@ -60,6 +60,14 @@ function create_jenkins_config_from_templates() {
 }
 
 function install_plugins() {
+  mkdir -p ${JENKINS_HOME}/plugins
+  
+  # with the /usr/lib64/jenkins -> /usr/lib/jenkins symlink, don't need to check both directories
+  if [ "$(ls -A /usr/lib/jenkins 2>/dev/null)" ]; then
+      echo "Copying $(ls /usr/lib/jenkins | wc -l) RPM installed Jenkins plugins to ${JENKINS_HOME} ..."
+      for FILENAME in /usr/lib/jenkins/*hpi ; do cp --remove-destination $FILENAME ${JENKINS_HOME}/plugins/`basename $FILENAME .hpi`.jpi; done
```

Comment:
I don't feel like I understand why --remove-destination is used here, especially as we're doing the lib/lib64 symlink now.  A code comment would be handy.

Reply:
it stemmed from my testing, and when files in JENKINS_HOME/plugins are links to begin with i.e. we have a PV and this is an upgrade scenario - the new sym link we are establishing has no bearing - if you copy a file to a new destination, where that destination is a link to another file, you get `cp: not writing through dangling symlink ...` ... see https://github.com/openshift/jenkins/pull/425#issuecomment-348321117 and https://github.com/openshift/jenkins/pull/425#issuecomment-348325086

I'll add a comment to try and illustrate / clarify